app/helpers/project_helper.rb

```diff
@@ -1,75 +1,78 @@
+  def date_select_options(url_params, possible_aggregation)
     Event::DATE_OPTIONS.map do |k, v|
       selected = "selected" if k == url_params[:date]
-      "<option data-url='#{build_date_url(k, url_params.permit(*Project::PROJECT_PARAMS), day_not_allowed)}' data-name='#{k}' #{selected}>#{v}</option>"
+      "<option data-url='#{build_date_url(k, url_params.permit(*Project::PROJECT_PARAMS), possible_aggregation)}' data-name='#{k}' #{selected}>#{v}</option>"
     end
   end
 
-  def build_event_url(event_name, url_params, day_not_allowed)
+  def build_event_url(event_name, url_params,possible_aggregation)
     project_path(
       url_params[:slug],
       url_params.except(:prop, :slug)
         .merge({ event: event_name.parameterize })
-        .merge(agg_clamp(url_params[:agg], day_not_allowed))
+        .merge(agg_clamp(url_params[:agg], possible_aggregation))
     )
   end
 
-  def build_agg_url(agg, url_params, day_not_allowed)
+  def build_agg_url(agg, url_params, possible_aggregation)
     project_path(
       url_params[:slug],
-      url_params.except(:slug).merge(agg_clamp(agg, day_not_allowed))
+      url_params.except(:slug).merge(agg_clamp(agg, possible_aggregation))
     )
   end
 
-  def build_property_url(prop, url_params, day_not_allowed)
+  def build_property_url(prop, url_params, possible_aggregation)
     project_path(
       url_params[:slug],
-      url_params.except(:slug).merge({ prop: prop }).merge(agg_clamp(url_params[:agg], day_not_allowed))
+      url_params.except(:slug).merge({ prop: prop }).merge(agg_clamp(url_params[:agg], possible_aggregation))
     )
   end
 
-  def build_date_url(date, url_params, day_not_allowed)
+  def build_date_url(date, url_params, possible_aggregation)
     project_path(
       url_params[:slug],
-      url_params.except(:slug).merge({ date: date }).merge(agg_clamp(url_params[:agg], day_not_allowed))
+      url_params.except(:slug).merge({ date: date }).merge(agg_clamp(url_params[:agg], possible_aggregation))
     )
   end
 
-  def build_test_toggle_url(test, url_params, day_not_allowed)
+  def build_test_toggle_url(test, url_params, possible_aggregation)
     url_params = url_params.permit(*Project::PROJECT_PARAMS)
     project_path(
       url_params[:slug],
-      url_params.except(:slug).merge({ test: test }).merge(agg_clamp(url_params[:agg], day_not_allowed))
+      url_params.except(:slug).merge({ test: test }).merge(agg_clamp(url_params[:agg], possible_aggregation))
     )
   end
 
-  def agg_clamp(agg, day_not_allowed)
-    { agg: day_not_allowed && agg == "d" ? "w" : agg }
+  def agg_clamp(agg, possible_aggregation)
+    { agg: !possible_aggregation.include?("d") && agg == "d" ? "w" : agg }
+    { agg: !possible_aggregation.include?("m") && agg == "m" ? "w" : agg }
+    { agg: !possible_aggregation.include?("m") && !possible_aggregation.include?("y") && agg == "y" ? "w" : agg }
+    { agg: !possible_aggregation.include?("y") && agg == "y" ? "m" : agg }
```

Comment:
Do we still need the `agg_clamp` function? Can't we just use the `agg` value directly?

Reply:
removed agg_clamp (.merge(agg_clamp(...)) and replaced it with the calculated aggregation parameter directly (.merge({ agg: aggregation) })